mantidimaging/core/parallel/shared.py

```diff
@@ -2,81 +2,12 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 from __future__ import annotations
 
-from functools import partial
-from typing import Any, TYPE_CHECKING
+from typing import Any
 from collections.abc import Callable
+from numpy import ndarray  # Import ndarray type from numpy
```

Comment:
This should stay inside the `if TYPE_CHECKING:` if `ndarray` is only used here for type annotations